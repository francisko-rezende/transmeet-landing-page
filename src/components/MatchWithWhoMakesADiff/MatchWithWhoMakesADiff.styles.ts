import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  margin-top: 26rem;
  display: grid;
  grid-template-columns: repeat(2, 1fr);

  @media (max-width: 809px) {
    margin-top: 17rem;
    grid-template-columns: 1fr;
    row-gap: 2rem;
  }
`
export const TextWrapper = styled.div`
  padding-left: ${({ theme }) => theme.grid.gutter};

  @media (max-width: 809px) {
    padding-left: unset;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`

export const ImgWrapper = styled.div`
  transform: translate(-10%, -20%);

  @media (max-width: 809px) {
    transform: unset;
    display: flex;
    justify-content: center;
  }
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.typographyColors.black};
    font-size: 5.2rem;
  `}

  @media (max-width: 809px) {
    text-align: center;
    font-size: 4.2rem;
  }
`

export const Paragraph = styled.p`
  line-height: 1.5;
  margin-top: 2.4rem;
  font-size: 2.1rem;

  ${({ theme }) => css`
    color: ${theme.colors.typographyColors.gray};
    font-weight: ${theme.font.weight.regular};
  `}

  @media (max-width: 809px) {
    text-align: center;
    font-size: 1.8rem;
  }
`

export const SecondaryText = styled.span`
  margin-top: 2rem;
  display: block;

  ${({ theme }) => css`
    color: ${theme.colors.secondaryColor.secondary};
    font-weight: ${theme.font.weight.regular};
    font-size: 1.6rem;
  `}

  @media (max-width: 809px) {
    text-align: center;
  }
`

export const BoldText = styled.span`
  ${({ theme }) => css`
    font-weight: ${theme.font.weight.bold};
    color: ${theme.colors.typographyColors.black};
  `}
`
