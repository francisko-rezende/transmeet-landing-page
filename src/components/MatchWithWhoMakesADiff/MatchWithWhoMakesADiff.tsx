import React from 'react'

import MainContainer from 'components/MainContainer'
import Image from 'next/image'

import * as S from './MatchWithWhoMakesADiff.styles'

const MatchWithWhoMakesADiff = () => {
  return (
    <MainContainer>
      <S.Wrapper>
        <S.ImgWrapper>
          <Image
            src="/images/pic-02.webp"
            width={547}
            height={509}
            alt="Estudante sorridente olhando o celular"
          />
        </S.ImgWrapper>
        <S.TextWrapper>
          <S.Title>Combine com quem faz a diferença!</S.Title>
          <S.Paragraph>
            Através da nossa plataforma é possível encontrar empresas que estão
            dispostas a patrocinar os estudos da população trans, travesti e não
            binária em diversas áreas do conhecimento.
          </S.Paragraph>
          <p>
            <S.SecondaryText>Nada de letrinha pequena...</S.SecondaryText>
            <S.BoldText>Escolheu, combinou, estudou!</S.BoldText>
          </p>
        </S.TextWrapper>
      </S.Wrapper>
    </MainContainer>
  )
}

export default MatchWithWhoMakesADiff
