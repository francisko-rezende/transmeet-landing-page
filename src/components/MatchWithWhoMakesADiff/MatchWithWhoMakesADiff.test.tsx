import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import MatchWithWhoMakesADiff from '.'

describe('<MatchWithWhoMakesADiff />', () => {
  test('should render the section with its different elements properly', () => {
    const { container } = renderWithTheme(<MatchWithWhoMakesADiff />)

    expect(
      screen.getByRole('img', {
        name: /estudante sorridente olhando o celular/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /combine com quem faz a diferença!/i
      })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', {
        name: /combine com quem faz a diferença!/i
      })
    ).toHaveStyle({ color: '#152536', fontSize: '5.2rem' })

    expect(
      screen.getByText(
        /através da nossa plataforma é possível encontrar empresas que estão dispostas a patrocinar os estudos da população trans, travesti e não binária em diversas áreas do conhecimento\./i
      )
    ).toBeInTheDocument()
    expect(
      screen.getByText(
        /através da nossa plataforma é possível encontrar empresas que estão dispostas a patrocinar os estudos da população trans, travesti e não binária em diversas áreas do conhecimento\./i
      )
    ).toHaveStyle({ color: '#68717A', fontSize: '2.1rem' })

    expect(
      screen.getByText(/nada de letrinha pequena\.\.\./i)
    ).toBeInTheDocument()
    expect(screen.getByText(/nada de letrinha pequena\.\.\./i)).toHaveStyle({
      color: '#BB4981',
      fontSize: '1.6rem'
    })

    expect(
      screen.getByText(/escolheu, combinou, estudou!/i)
    ).toBeInTheDocument()
    expect(screen.getByText(/escolheu, combinou, estudou!/i)).toHaveStyle({
      fontWeight: '700',
      color: '#152536'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
