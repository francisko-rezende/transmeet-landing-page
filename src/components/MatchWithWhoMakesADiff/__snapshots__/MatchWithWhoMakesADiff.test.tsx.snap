// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<MatchWithWhoMakesADiff /> should render the section with its different elements properly 1`
] = `
.c0 {
  max-width: calc(1170px + 44px);
  padding: 0 22px;
  margin: 0 auto;
}

.c1 {
  margin-top: 26rem;
  display: grid;
  grid-template-columns: repeat(2,1fr);
}

.c3 {
  padding-left: 3rem;
}

.c2 {
  -webkit-transform: translate(-10%,-20%);
  -ms-transform: translate(-10%,-20%);
  transform: translate(-10%,-20%);
}

.c4 {
  color: #152536;
  font-size: 5.2rem;
}

.c5 {
  line-height: 1.5;
  margin-top: 2.4rem;
  font-size: 2.1rem;
  color: #68717A;
  font-weight: 400;
}

.c6 {
  margin-top: 2rem;
  display: block;
  color: #BB4981;
  font-weight: 400;
  font-size: 1.6rem;
}

.c7 {
  font-weight: 700;
  color: #152536;
}

@media (max-width:809px) {
  .c1 {
    margin-top: 17rem;
    grid-template-columns: 1fr;
    row-gap: 2rem;
  }
}

@media (max-width:809px) {
  .c3 {
    padding-left: unset;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:809px) {
  .c2 {
    -webkit-transform: unset;
    -ms-transform: unset;
    transform: unset;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:809px) {
  .c4 {
    text-align: center;
    font-size: 4.2rem;
  }
}

@media (max-width:809px) {
  .c5 {
    text-align: center;
    font-size: 1.8rem;
  }
}

@media (max-width:809px) {
  .c6 {
    text-align: center;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27547%27%20height=%27509%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Estudante sorridente olhando o celular"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Combine com quem faz a diferença!
      </h2>
      <p
        class="c5"
      >
        Através da nossa plataforma é possível encontrar empresas que estão dispostas a patrocinar os estudos da população trans, travesti e não binária em diversas áreas do conhecimento.
      </p>
      <p>
        <span
          class="c6"
        >
          Nada de letrinha pequena...
        </span>
        <span
          class="c7"
        >
          Escolheu, combinou, estudou!
        </span>
      </p>
    </div>
  </section>
</div>
`
